### PR TITLE
Passive declare of exchanges and queues

### LINKED ISSUE #33: Support passive declarations of queues and exchanges
I'm trying to accomplish:
> Publish to exchange `X` if exists, if not do nothing.

First, I used `passive=True`in `asynqp.spec.ExchangeDeclare`, which produces (as expected) an error in RabbitMQ:
```bash
=ERROR REPORT==== 26-Aug-2015::11:12:47 ===
Channel error on connection <0.1206.0> (127.0.0.1:47403 -> 127.0.0.1:5672, vhost: '/', user: 'guest'), channel 2:
{amqp_error,not_found,"no exchange 'exchange.user.cccc' in vhost '/'",'exchange.declare'}
```
But nothing on python application side:
```python
rmq_chan.sender.send_method(asynqp.spec.ExchangeDeclare(
    0, 
    'exchange.user.cccc',  # name
    'fanout',              # type
    True,                  # passive
    False,                 # durable
    True,                  
    False, 
    # False,               # no_wait  
    {}                     # arguments
))
```
But why bother if we can just publish to non existent exchange with the same effect:
```python
rmq_chan.sender.send_BasicPublish(
    'exchange.user.cccc',
    routing_key='',
    mandatory=False,
    message=asynqp.Message('{"zzz":33}')
)
```
Again, no error in application despite error in RabbitMQ.
```bash
=ERROR REPORT==== 26-Aug-2015::10:53:17 ===
Channel error on connection <0.1206.0> (127.0.0.1:47403 -> 127.0.0.1:5672, vhost: '/', user: 'guest'), channel 1:
{amqp_error,not_found,"no exchange 'exchange.user.cccc' in vhost '/'",'basic.publish'}
```
This would not be a big deal except that `rmq_chan` cannot be used to send any more commands to RabbitMQ after such error (and I don't actually know it happened).

### PR DESCRIPTION
**Pull request by Drizzt1991**
Fixes #33

### CODE REVIEW

File: test/exchange_tests.py [hunk 1]
```diff
@@ -174,3 +175,46 @@ def when_confirmation_arrives(self):
...
+            self.channel.declare_exchange(
+                'name_1', 'fanout', passive=True,
+                durable=True, auto_delete=False, internal=False))
+        self.tick()
+
+    def when_error_arrives(self):
+        self.server.send_method(
+            self.channel.id, spec.ChannelClose(404, 'Bad exchange', 40, 50))
+
+    def it_should_return_an_exchange_object(self):
```
Comment (benjamin-hodgson) [inline test/exchange_tests.py]: Misleading method name :wink:

File: test/exchange_tests.py [hunk 2]
```diff
@@ -174,3 +175,46 @@ def when_confirmation_arrives(self):
...
+        try:
+            self.task.result()
+        except exceptions.NotFound:
+            pass
+        else:
+            assert False, "NotFound exception not raised"
```
Comment (benjamin-hodgson) [inline test/exchange_tests.py]: `assert isinstance(self.task.exception(), exceptions.NotFound)` is cleaner. https://docs.python.org/3/library/asyncio-task.html#asyncio.Future.exception

File: test/queue_tests.py [hunk 3]
```diff
@@ -412,3 +413,45 @@ def when_connection_is_closed(self):
...
+        self.server.send_method(
+            self.channel.id, spec.ChannelClose(404, 'Bad queue', 40, 50))
+
+    def it_should_raise_exception(self):
+        try:
+            self.task.result()
+        except exceptions.NotFound:
+            pass
+        else:
+            assert False, "NotFound exception not raised"
```
Comment (benjamin-hodgson) [inline test/queue_tests.py]: [same here](https://github.com/benjamin-hodgson/asynqp/pull/61/files#r43736282)

**Comment by benjamin-hodgson** [comment]
This is definitely something I want to support. I think this recent work on the error handling infrastructure constitutes the downpayment which will get us good features like this. Thanks!

**Comment by Drizzt1991** [comment]
Fixed

**Pull request merged**